components/server/src/oauth-server/oauth-controller.ts

```diff
@@ -175,6 +176,17 @@ export class OAuthController {
             }
         });
 
+        router.get("/oauth/inspect", async (req: express.Request, res: express.Response) => {
+            const clientId = req.query.client as string;
+            if (typeof clientId !== "string" || !Object.keys(inMemoryDatabase.clients).includes(clientId)) {
```

Comment:
this definitely has the potential to provide an unguarded endpoint to scan for random `clientId`s.

tagging as security relevant.

Reply:
everything is already available in this repo? 🤔

Reply:
yeah client ids are already public

Reply:
I think the point Alex is making is that this allows you to enumerate all client IDs we may have. Therefore, you can estimate how big of an installation of gitpod this is, how many users it has and which exact IDs exist in the system. 

There's a difference between the client ID being public and knowing exactly how many client IDs there are.

Reply:
What would be an alternative? cc @geropl

Reply:
> Therefore, you can estimate how big of an installation of gitpod this is, how many users it has and which exact IDs exist in the system.

I'm not sure how you can do it though. You can only there Gitpod installation supports installing ssh keys or not. You cannot get any other information from this endpoint.

Reply:
Would you be able to provide a use-case this is looking to solve? Is it to discover my own ClientID?

Reply:
Please have a look at this [RFC](https://www.notion.so/gitpod/Gitpod-Server-API-compatibility-check-ee00fcfe60a84c1e8e018ab6c06c1f5d#5aed369a61cf46c7bf945ae017a6e125) It is a work around for how we do auth now.

Reply:
@mikenikles  @AlexTugarev  I think there is a misunderstanding here: Clients !== API clients.

Client is a possible _kind_ of clients we support as defined [here](https://github.com/gitpod-io/gitpod/blob/79ef436f154158d4a4c78aa7e619cedfd52ee83d/components/server/src/oauth-server/db.ts#L38-L96).

> everything is already available in this repo?

:point_up: Enumerating those is fine I think. :slightly_smiling_face: